### ISSUE TITLE
[HxGrid] Ability to reset grid state (position?) when refreshing data (e.g. after new filter applied)

### ISSUE THREAD

**Issue by hakenr** [havit/Havit.Blazor#415]
We should give the developers an easy way to navigate the beginning of the grid after new filtering criteria are applied (no matter whether it is in paging, inifinite-scroll or any other future mode). Maybe something like (new optional parameter)
```csharp
await gridComponent.RefreshDataAsync(resetUserState: true);
```
...which will be used when the Apply button in filter is clicked?

Do we also want to reset sorting or just the position?

(extracted from #412)

**Comment by hakenr** [comment]
From discussion:
* we want to jump to start (is it achievable with `Virtualize`? new instance?),
* we want to keep the sorting,
* we want to keep other potential grid settings (column width, column visibility, ...)

**Comment by hakenr** [comment]
@Harvey1214 This is still marked as `#TBD`, i.e., not specified enough to start the implementation. Let me know if you need this functionality for any of your scenarios, and we can move forward.

**Comment by ericwscott** [comment]
@hakenr is it currently possible to reset paging position when the dataset changes?  This task is to provide an easy way, but are there any workarounds?

**Comment by hakenr** [comment]
@ericwscott The current workaround is to use ` @bind-CurrentUserState="_userState"` and reset the `PageIndex` in `_userState` field.

**Comment by gdube-nrc** [comment]
Just want to point out that this workaround doesn't work for a virtualized grid, can you confirm?

**Comment by hakenr** [comment]
@gdube-nrc Correct, restoring the grid position in a virtualized grid isn't supported. The virtualization depends on the underlying `Virtualize` Blazor component, which doesn't offer this feature.

**Comment by gdube-nrc** [comment]
Okay, thank you. Then I guess the only way would be to use JS Interop to reset the scroll position manually.
